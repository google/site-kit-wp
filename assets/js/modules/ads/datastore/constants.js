/**
 * `modules/ads` data store constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_ADS = 'modules/ads';

export const MODULE_SLUG_ADS = 'ads';

export const ADWORDS_SCOPE = 'https://www.googleapis.com/auth/adwords';
export const SUPPORT_CONTENT_SCOPE =
	'https://www.googleapis.com/auth/supportcontent';

// Date range offset days for Ads report requests.
export const DATE_RANGE_OFFSET = 1;

export const ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY =
	'ads_module_setup_banner_prompt_dismissed_key';

export const ADS_WOOCOMMERCE_REDIRECT_MODAL_CACHE_KEY = 'wc-redirect-modal';

export const PLUGINS = {
	WOOCOMMERCE: 'woocommerce',
	GOOGLE_FOR_WOOCOMMERCE: 'google-listings-and-ads',
};

export const AVAILABLE_PLUGINS = [
	PLUGINS.WOOCOMMERCE,
	PLUGINS.GOOGLE_FOR_WOOCOMMERCE,
];
