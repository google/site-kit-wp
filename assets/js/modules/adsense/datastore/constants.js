/**
 * `modules/adsense` data store constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_ADSENSE = 'modules/adsense';

export const MODULE_SLUG_ADSENSE = 'adsense';

// Date range offset days for AdSense report requests.
export const DATE_RANGE_OFFSET = 1;

export const API_STATE_READY = 'READY';
export const API_STATE_NEEDS_ATTENTION = 'NEEDS_ATTENTION';
export const API_STATE_REQUIRES_REVIEW = 'REQUIRES_REVIEW';
export const API_STATE_GETTING_READY = 'GETTING_READY';

// State to determine whether background submission is allowable.
export const BACKGROUND_SUBMIT_SUSPENDED = 'background-submit-suspended';

export const AD_BLOCKING_FORM_SETTINGS = 'adsenseAdBlockingFormSettings';

export const AD_BLOCKING_RECOVERY_SETUP_CREATE_MESSAGE_CTA_CLICKED =
	'googlesitekit-ad-blocking-recovery-setup-create-message-cta-clicked';

export const AD_BLOCKING_RECOVERY_MAIN_NOTIFICATION_KEY =
	'ad-blocking-recovery-notification';

// Various ad blocking recovery setup statuses.
export const ENUM_AD_BLOCKING_RECOVERY_SETUP_STATUS = {
	TAG_PLACED: 'tag-placed',
	SETUP_CONFIRMED: 'setup-confirmed',
};

// Zero-based index of the available ad blocking recovery setup steps.
export const ENUM_AD_BLOCKING_RECOVERY_SETUP_STEP = {
	PLACE_TAGS: 0,
	CREATE_MESSAGE: 1,
	COMPLETE: 2,
};
