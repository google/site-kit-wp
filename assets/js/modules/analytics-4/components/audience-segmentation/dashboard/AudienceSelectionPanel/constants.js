/**
 * AudienceSelectionPanel components - constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const AUDIENCE_ADD_GROUP_NOTICE_SLUG =
	'audience-segmentation-add-group-notice';
export const AUDIENCE_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-audience-selection-panel-opened';
export const AUDIENCE_SELECTION_FORM = 'audience-selection-form';
export const AUDIENCE_SELECTED = 'audience-selected';
export const AUDIENCE_SELECTION_CHANGED = 'audience-selection-changed';
export const AUDIENCE_CREATION_NOTICE_SLUG =
	'audience-segmentation-creation-notice';
export const AUDIENCE_CREATION_SUCCESS_NOTICE_SLUG =
	'audience-segmentation-creation-success-notice';
export const MIN_SELECTED_AUDIENCES_COUNT = 1;
export const MAX_SELECTED_AUDIENCES_COUNT = 3;
export const AUDIENCE_CREATION_EDIT_SCOPE_NOTICE_SLUG =
	'audience-creation-edit-scope-notice';
export const AUDIENCE_CREATION_FORM = 'audience-creation-form';
