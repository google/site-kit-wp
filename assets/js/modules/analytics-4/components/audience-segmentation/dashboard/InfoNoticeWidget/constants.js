/**
 * Constants related to audience segmentation info notices.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * UI key to display audience segmentation info notice.
 *
 * @since 1.135.0
 * @private
 */
export const AUDIENCE_INFO_NOTICE_HIDE_UI =
	'audience-segmentation-info-notice-ui';

/**
 * Slug for audience segmentation info notice.
 *
 * @since 1.130.0
 * @private
 */
export const AUDIENCE_INFO_NOTICE_SLUG = 'audience-segmentation-info-notice';

// Notice slugs: new-visitors, compare-metrics, custom-audiences, purchasers, returning-visitors, compare-new-returning, or compare-cities

/**
 * List of info notices for audiences.
 *
 * @since 1.130.0
 * @private
 */
export const AUDIENCE_INFO_NOTICES = [
	{
		slug: 'new-visitors',
		content: __(
			'The higher the portion of new visitors you have, the more your audience is growing. Looking at what content brings them to your site may give you insights on how to reach even more people.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-metrics',
		content: __(
			'Select up to three visitor groups to display on the dashboard and easily compare metrics between them.',
			'google-site-kit'
		),
	},
	{
		slug: 'custom-audiences',
		content: __(
			'Configure your own custom audiences in Analytics to gain deeper insights into visitor behavior, for example consider creating an “Existing customers” or “Subscribers” segment, depending on what goals you have for your site.',
			'google-site-kit'
		),
	},
	{
		slug: 'purchasers',
		content: __(
			'Select the Purchasers visitor group to gain insights into which visitors bring the most revenue to your site.',
			'google-site-kit'
		),
	},
	{
		slug: 'returning-visitors',
		content: __(
			'The more returning visitors your site has, the stronger and more loyal an audience you’re building. Check which content brings people back to your site - it might help you create a strategy to build a community.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-new-returning',
		content: __(
			'Compare the ratio of “new” to “returning” visitors – this can give you insights on whether you have more people stopping by as a one-off, or more loyal visitors.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-cities',
		content: __(
			'Check the cities which bring you more new vs more returning visitors – there might be new audiences you could engage with in locations you hadn’t thought about.',
			'google-site-kit'
		),
	},
];
