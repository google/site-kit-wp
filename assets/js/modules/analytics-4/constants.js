/**
 * GA4 module constants.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const ACTIVATION_STEP_REMINDER = 0;
export const ACTIVATION_STEP_SETUP = 1;
export const ACTIVATION_STEP_SUCCESS = 2;
export const ACTIVATION_STEP_IN_PROGRESS = 3;

export const ACTIVATION_ACKNOWLEDGEMENT_TOOLTIP_STATE_KEY =
	'activation-acknowledgement-tooltip-state';

export const GA4_ACTIVATION_BANNER_STATE_KEY = 'ga4-activation-banner-state';

export const ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY =
	'enhanced-measurement-activation-banner-tooltip-state';
export const LEGACY_ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY =
	'enhanced-measurement-activation-banner-dismissed-item';

export const REPORT_ARGS_SELECTED_METRIC_KEY = '_r.explorerCard..selmet';
export const REPORT_ARGS_SELECTED_DIMENSION_KEY = '_r.explorerCard..seldim';
export const REPORT_ARGS_DATA_FILTERS_KEY = '_r..dataFilters';
export const REPORT_ARGS_NAV_KEY = '_r..nav';

export const KM_CONNECT_GA4_CTA_WIDGET_DISMISSED_ITEM_KEY =
	'key-metrics-connect-ga4-cta-widget';

/**
 * Date that Site Kit will automatically switch to using GA4: September 25, 2023.
 *
 * With Site Kit's maximum date range being three months, once this period has passed since the UA cutoff date,
 * it will no longer be able to show any historical data for UA and will automatically switch to GA4.
 * For operational reasons, we automatically make the switch a few days before the three month cutoff point.
 *
 * @since 1.107.0
 * @see UA_CUTOFF_DATE.
 */
export const GA4_AUTO_SWITCH_DATE = '2023-09-25';
export const MODULE_SLUG_ANALYTICS_4 = 'analytics-4';
