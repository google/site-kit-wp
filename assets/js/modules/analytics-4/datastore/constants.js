/**
 * `modules/analytics-4` data store constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_ANALYTICS_4 = 'modules/analytics-4';

// A special Account ID value used for the "Set up a new account" option.
export const ACCOUNT_CREATE = 'account_create';

// A special Property ID value used for the "Set up a new property" option.
export const PROPERTY_CREATE = 'property_create';

export const WEBDATASTREAM_CREATE = 'webdatastream_create';

export const FORM_SETUP = 'analyticsSetup';

export const MAX_WEBDATASTREAMS_PER_BATCH = 10;

// Date range offset days for Analytics 4 report requests.
export const DATE_RANGE_OFFSET = 1;

export const GTM_SCOPE = 'https://www.googleapis.com/auth/tagmanager.readonly';

export const ENHANCED_MEASUREMENT_FORM = 'enhanced-measurement-form';
export const ENHANCED_MEASUREMENT_ENABLED = 'enhanced-measurement-enabled';
export const ENHANCED_MEASUREMENT_SHOULD_DISMISS_ACTIVATION_BANNER =
	'enhanced-measurement-should-dismiss-activation-banner';

// Form ID for the account creation form.
export const FORM_ACCOUNT_CREATE = 'analyticsAccountCreate';

export const FORM_CUSTOM_DIMENSIONS_CREATE = 'analyticsCustomDimensionsCreate';

// OAuth scope required for provisioning a Google Analytics account.
export const PROVISIONING_SCOPE =
	'https://www.googleapis.com/auth/analytics.provision';
export const EDIT_SCOPE = 'https://www.googleapis.com/auth/analytics.edit';

// Dashboard widget constants.
export const UI_DIMENSION_NAME = 'dashboardAllTrafficWidgetDimensionName';
export const UI_DIMENSION_COLOR = 'dashboardAllTrafficWidgetDimensionColor';
export const UI_DIMENSION_VALUE = 'dashboardAllTrafficWidgetDimensionValue';
export const UI_ACTIVE_ROW_INDEX = 'dashboardAllTrafficWidgetActiveRowIndex';
export const UI_ALL_TRAFFIC_LOADED = 'dashboardAllTrafficWidgetLoaded';

// Note: names and descriptions are not translated as these are not surfaced in Site Kit
// and are also subject to hard limits on the length which would be unpredictable if translated.
// See https://developers.google.com/analytics/devguides/config/admin/v1/rest/v1beta/properties.customDimensions#CustomDimension
export const CUSTOM_DIMENSION_DEFINITIONS = {
	googlesitekit_post_date: {
		parameterName: 'googlesitekit_post_date',
		displayName: 'WordPress Post Date',
		description: 'Created by Site Kit: Date when a post was published',
		scope: 'EVENT',
	},
	googlesitekit_post_author: {
		parameterName: 'googlesitekit_post_author',
		displayName: 'WordPress Post Author',
		description: 'Created by Site Kit: WordPress name of the post author',
		scope: 'EVENT',
	},
	googlesitekit_post_categories: {
		parameterName: 'googlesitekit_post_categories',
		displayName: 'WordPress Post Categories',
		description:
			'Created by Site Kit: Names of categories assigned to a post',
		scope: 'EVENT',
	},
	googlesitekit_post_type: {
		parameterName: 'googlesitekit_post_type',
		displayName: 'WordPress Post Type',
		description: 'Created by Site Kit: Content type of a post',
		scope: 'EVENT',
	},
};

export const ENUM_CONVERSION_EVENTS = {
	ADD_TO_CART: 'add_to_cart',
	PURCHASE: 'purchase',
	SUBMIT_LEAD_FORM: 'submit_lead_form',
	GENERATE_LEAD: 'generate_lead',
	CONTACT: 'contact',
};
export const CONVERSION_REPORTING_LEAD_EVENTS = [
	ENUM_CONVERSION_EVENTS.CONTACT,
	ENUM_CONVERSION_EVENTS.GENERATE_LEAD,
	ENUM_CONVERSION_EVENTS.SUBMIT_LEAD_FORM,
];

// Audience enums.
export const AUDIENCE_FILTER_CLAUSE_TYPE_ENUM = {
	AUDIENCE_CLAUSE_TYPE_UNSPECIFIED: 'AUDIENCE_CLAUSE_TYPE_UNSPECIFIED',
	INCLUDE: 'INCLUDE',
	EXCLUDE: 'EXCLUDE',
};

export const AUDIENCE_FILTER_SCOPE_ENUM = {
	AUDIENCE_FILTER_SCOPE_UNSPECIFIED: 'AUDIENCE_FILTER_SCOPE_UNSPECIFIED',
	AUDIENCE_FILTER_SCOPE_WITHIN_SAME_EVENT:
		'AUDIENCE_FILTER_SCOPE_WITHIN_SAME_EVENT',
	AUDIENCE_FILTER_SCOPE_WITHIN_SAME_SESSION:
		'AUDIENCE_FILTER_SCOPE_WITHIN_SAME_SESSION',
	AUDIENCE_FILTER_SCOPE_ACROSS_ALL_SESSIONS:
		'AUDIENCE_FILTER_SCOPE_ACROSS_ALL_SESSIONS',
};

export const SITE_KIT_AUDIENCE_DEFINITIONS = {
	'new-visitors': {
		description: 'People who visited the site for the first time',
		displayName: 'New visitors',
		membershipDurationDays: -1, // The longest duration, 540 days.
		filterClauses: [
			{
				clauseType: 'INCLUDE',
				simpleFilter: {
					scope: 'AUDIENCE_FILTER_SCOPE_ACROSS_ALL_SESSIONS',
					filterExpression: {
						andGroup: {
							filterExpressions: [
								{
									orGroup: {
										filterExpressions: [
											{
												dimensionOrMetricFilter: {
													fieldName: 'newVsReturning',
													stringFilter: {
														matchType: 'EXACT',
														value: 'new',
													},
												},
											},
										],
									},
								},
								{
									orGroup: {
										filterExpressions: [
											{
												notExpression: {
													dimensionOrMetricFilter: {
														fieldName: 'groupId',
														stringFilter: {
															matchType: 'EXACT',
															value: 'created_by_googlesitekit:new_visitors',
														},
													},
												},
											},
										],
									},
								},
							],
						},
					},
				},
			},
		],
	},
	'returning-visitors': {
		description: 'People who have visited your site at least once before',
		displayName: 'Returning visitors',
		membershipDurationDays: -1, // The longest duration, 540 days.
		filterClauses: [
			{
				clauseType: 'INCLUDE',
				simpleFilter: {
					scope: 'AUDIENCE_FILTER_SCOPE_ACROSS_ALL_SESSIONS',
					filterExpression: {
						andGroup: {
							filterExpressions: [
								{
									orGroup: {
										filterExpressions: [
											{
												dimensionOrMetricFilter: {
													fieldName: 'newVsReturning',
													stringFilter: {
														matchType: 'EXACT',
														value: 'returning',
													},
												},
											},
										],
									},
								},
								{
									orGroup: {
										filterExpressions: [
											{
												notExpression: {
													dimensionOrMetricFilter: {
														fieldName: 'groupId',
														stringFilter: {
															matchType: 'EXACT',
															value: 'created_by_googlesitekit:returning_visitors',
														},
													},
												},
											},
										],
									},
								},
							],
						},
					},
				},
			},
		],
	},
};

export const AUDIENCE_SEGMENTATION_SETUP_FORM = 'audiencePermissionsSetup';

export const AUDIENCE_TILE_CUSTOM_DIMENSION_CREATE =
	'audienceTileCustomDimensionCreate';

export const AUDIENCE_TYPE_SORT_ORDER = {
	USER_AUDIENCE: 0,
	SITE_KIT_AUDIENCE: 1,
	DEFAULT_AUDIENCE: 2,
};

export const AUDIENCE_ITEM_NEW_BADGE_SLUG_PREFIX =
	'audience-selection-panel-expirable-new-badge-';
