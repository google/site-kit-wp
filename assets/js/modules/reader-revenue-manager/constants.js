/**
 * Reader Revenue Manager constants.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const SNIPPET_MODES = {
	post_types: __( 'Specific content types', 'google-site-kit' ),
	per_post: __( 'Specified pages', 'google-site-kit' ),
	sitewide: __( 'Site wide', 'google-site-kit' ),
};

export const RRM_SETUP_NOTIFICATION_ID = 'rrm-setup-notification';
export const RRM_SETUP_SUCCESS_NOTIFICATION_ID =
	'setup-success-notification-rrm';
export const RRM_PRODUCT_ID_CONTRIBUTIONS_NOTIFICATION_ID =
	'rrm-product-id-contributions-notification';
export const RRM_PRODUCT_ID_SUBSCRIPTIONS_NOTIFICATION_ID =
	'rrm-product-id-subscriptions-notification';
