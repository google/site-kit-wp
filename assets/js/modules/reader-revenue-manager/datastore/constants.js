/**
 * `modules/reader-revenue-manager` data store constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const ERROR_CODE_NON_HTTPS_SITE = 'non_https_site';

export const MODULES_READER_REVENUE_MANAGER = 'modules/reader-revenue-manager';

export const READER_REVENUE_MANAGER_MODULE_SLUG = 'reader-revenue-manager';

export const PUBLICATION_ONBOARDING_STATES = {
	ONBOARDING_COMPLETE: 'ONBOARDING_COMPLETE',
	ONBOARDING_ACTION_REQUIRED: 'ONBOARDING_ACTION_REQUIRED',
	PENDING_VERIFICATION: 'PENDING_VERIFICATION',
	UNSPECIFIED: 'ONBOARDING_STATE_UNSPECIFIED',
};

export const UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION =
	'READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION';

export const LEGACY_RRM_SETUP_BANNER_DISMISSED_KEY =
	'rrm_module_setup_banner_dismissed_key';

export const READER_REVENUE_MANAGER_SETUP_FORM =
	'readerRevenueManagerSetupForm';

export const READER_REVENUE_MANAGER_NOTICES_FORM =
	'readerRevenueManagerNoticesForm';

export const SHOW_PUBLICATION_CREATE = 'showPublicationCreate';

export const RESET_PUBLICATIONS = 'resetPublications';

export const SYNC_PUBLICATION = 'syncPublication';
