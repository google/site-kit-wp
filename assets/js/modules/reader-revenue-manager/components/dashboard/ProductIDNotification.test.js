/**
 * ProductIDNotification component tests.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import {
	createTestRegistry,
	provideModules,
	render,
} from '../../../../../../tests/js/test-utils';
import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
import { VIEW_CONTEXT_MAIN_DASHBOARD } from '../../../../googlesitekit/constants';
import { RRM_PRODUCT_ID_NOTIFICATION_ID } from '../../constants';
import {
	MODULES_READER_REVENUE_MANAGER,
	PUBLICATION_ONBOARDING_STATES,
} from '../../datastore/constants';
import { NOTIFICATIONS } from '../..';
import { withNotificationComponentProps } from '../../../../googlesitekit/notifications/util/component-props';
import ProductIDNotification from './ProductIDNotification';

describe( 'ProductIDNotification', () => {
	let registry;

	const notification = NOTIFICATIONS[ RRM_PRODUCT_ID_NOTIFICATION_ID ];

	const ProductIDNotificationComponent = withNotificationComponentProps(
		RRM_PRODUCT_ID_NOTIFICATION_ID
	)( ProductIDNotification );

	beforeEach( () => {
		registry = createTestRegistry();

		provideModules( registry, [
			{
				slug: 'reader-revenue-manager',
				active: true,
				connected: true,
			},
		] );

		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
	} );

	afterEach( () => {
		jest.clearAllMocks();
	} );

	it( 'should render correctly when payment option is `contributions`', () => {
		registry
			.dispatch( MODULES_READER_REVENUE_MANAGER )
			.receiveGetSettings( {
				paymentOption: 'contributions',
				productIDs: [],
				productID: null, // TODO: Check what to do about the default value of 'openaccess'.
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			} );

		const { container, getByText } = render(
			<ProductIDNotificationComponent />,
			{
				registry,
				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
			}
		);

		expect(
			getByText(
				'New! You can now select product IDs to use with your Reader Revenue Manager snippet'
			)
		).toBeInTheDocument();

		expect( container ).toMatchSnapshot();
	} );

	it( 'should render correctly when payment option is `subscriptions`', () => {
		registry
			.dispatch( MODULES_READER_REVENUE_MANAGER )
			.receiveGetSettings( {
				paymentOption: 'subscriptions',
				productIDs: [],
				productID: null, // TODO: Check what to do about the default value of 'openaccess'.
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			} );

		const { container, getByText } = render(
			<ProductIDNotificationComponent />,
			{
				registry,
				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
			}
		);

		expect(
			getByText(
				'To complete your Reader Revenue Manager paywall setup, add your product IDs in settings'
			)
		).toBeInTheDocument();

		expect( container ).toMatchSnapshot();
	} );

	describe( 'checkRequirements', () => {
		it( 'is active when all required conditions are met', async () => {
			registry
				.dispatch( MODULES_READER_REVENUE_MANAGER )
				.receiveGetSettings( {
					paymentOption: 'subscriptions',
					productIDs: [ '123', '456' ],
					productID: null, // TODO: Check what to do about the default value of 'openaccess'.
					publicationOnboardingState:
						PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
				} );

			const isActive = await notification.checkRequirements(
				registry,
				VIEW_CONTEXT_MAIN_DASHBOARD
			);

			expect( isActive ).toBe( true );
		} );
	} );

	it.each( [
		[
			'publicationOnboardingState is not ONBOARDING_COMPLETE',
			{
				paymentOption: 'contributions',
				productIDs: [ '123', '456' ],
				productID: null,
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_ACTION_REQUIRED,
			},
		],
		[
			'productIDs is empty',
			{
				paymentOption: 'contributions',
				productIDs: [],
				productID: null,
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			},
		],
		[
			'productID is not null',
			{
				paymentOption: 'contributions',
				productIDs: [ '123', '456' ],
				productID: '123',
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			},
		],
		[
			'paymentOption is not `contributions` or `subscriptions`',
			{
				paymentOption: 'openaccess',
				productIDs: [ '123', '456' ],
				productID: null,
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			},
		],
	] )( 'should not be active when %s', async ( _, settings ) => {
		registry
			.dispatch( MODULES_READER_REVENUE_MANAGER )
			.receiveGetSettings( settings );

		const isActive = await notification.checkRequirements(
			registry,
			VIEW_CONTEXT_MAIN_DASHBOARD
		);

		expect( isActive ).toBe( false );
	} );
} );
