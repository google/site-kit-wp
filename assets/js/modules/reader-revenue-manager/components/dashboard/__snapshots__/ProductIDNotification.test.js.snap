// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductIDNotification should render correctly when payment option is \`contributions\` 1`] = `
<div>
  <section
    id="rrm-product-id-notification"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-subtle-notification googlesitekit-subtle-notification--new-feature mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-12"
        >
          <div
            class="googlesitekit-subtle-notification__icon"
          >
            <svg />
          </div>
          <div
            class="googlesitekit-subtle-notification__content"
          >
            <p />
            <p
              class="googlesitekit-subtle-notification__secondary_description"
            >
              New! You can now select product IDs to use with your Reader Revenue Manager snippet
            </p>
          </div>
          <div
            class="googlesitekit-subtle-notification__action"
          >
            <button
              class="mdc-button mdc-button--tertiary"
              target="_self"
            >
              <span
                class="mdc-button__label"
              >
                Got it
              </span>
            </button>
            <a
              class="mdc-button googlesitekit-subtle-notification__cta mdc-button--raised"
              href="undefined#connected-services/reader-revenue-manager"
              role="button"
              target="_self"
            >
              <span
                class="mdc-button__label"
              >
                Edit settings
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ProductIDNotification should render correctly when payment option is \`subscriptions\` 1`] = `
<div>
  <section
    id="rrm-product-id-notification"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-subtle-notification googlesitekit-subtle-notification--warning mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-12"
        >
          <div
            class="googlesitekit-subtle-notification__icon"
          >
            <svg />
          </div>
          <div
            class="googlesitekit-subtle-notification__content"
          >
            <p />
            <p
              class="googlesitekit-subtle-notification__secondary_description"
            >
              To complete your Reader Revenue Manager paywall setup, add your product IDs in settings
            </p>
          </div>
          <div
            class="googlesitekit-subtle-notification__action"
          >
            <button
              class="mdc-button mdc-button--tertiary"
              target="_self"
            >
              <span
                class="mdc-button__label"
              >
                Got it
              </span>
            </button>
            <a
              class="mdc-button googlesitekit-subtle-notification__cta mdc-button--raised"
              href="undefined#connected-services/reader-revenue-manager"
              role="button"
              target="_self"
            >
              <span
                class="mdc-button__label"
              >
                Edit settings
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
